src/Microsoft.Azure.WebJobs.Host/Executors/FunctionExecutor.cs

```diff
@@ -475,9 +521,10 @@ private static ITaskSeriesTimer StartParameterLogTimer(IRecurrentCommand updateC
 
         internal static async Task ExecuteWithWatchersAsync(IFunctionInstance instance,
             IReadOnlyDictionary<string, IValueProvider> parameters,
-            TimeSpan? globalFunctionTimeout,
+            JobHostFunctionTimeoutConfiguration timeoutConfig,
```

Comment:
Your call on whether you want to name this with the "JobHost" prefix or remove that, as I've done with SingletonConfiguration